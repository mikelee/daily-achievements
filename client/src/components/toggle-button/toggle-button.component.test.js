import { shallow, configure }  from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallowToJson } from 'enzyme-to-json'
import React from 'react';
import ToggleButton from './toggle-button.component';

configure({ adapter: new Adapter() });

it('should render ToggleButton with CheckRoundedIcon', () => {
    const wrapper = shallow(<ToggleButton type='forTodo' completed={1} toggleTodoCompleted={jest.fn()} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('should render Toggle Button without CheckRoundedIcon', () => {
    const wrapper = shallow(<ToggleButton type='forTodo' completed={0} toggleTodoCompleted={jest.fn()} />)

    expect(shallowToJson(wrapper)).toMatchSnapshot();
})