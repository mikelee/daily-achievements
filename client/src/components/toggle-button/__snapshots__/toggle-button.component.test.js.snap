// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Toggle Button without CheckRoundedIcon 1`] = `
<div
  className="toggle-button-container"
>
  <button
    className="toggle-button "
    onClick={[MockFunction]}
  />
</div>
`;

exports[`should render ToggleButton with CheckRoundedIcon 1`] = `
<div
  className="toggle-button-container"
>
  <button
    className="toggle-button "
    onClick={[MockFunction]}
  >
    <CheckRoundedIcon
      className="check-icon "
      fontSize="large"
    />
  </button>
</div>
`;
